sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala

```diff
@@ -140,11 +141,19 @@ case class UnresolvedDBObjectName(nameParts: Seq[String], isNamespace: Boolean)
   override def output: Seq[Attribute] = Nil
 }
 
+/**
+ * A resolved leaf like node that its statistics is no meaning.
+ */
+trait LeafNodeWithoutStats extends LogicalPlan with LeafLike[LogicalPlan] {
```

Comment:
Can we make it an abstract class and extends `LeafNode`? I'm a bit worried about changing the class hierarchy unnecessarily in this PR.

Reply:
changed to `trait LeafNodeWithoutStats extend LeafNode`